src/dune_rules/action_to_sh.ml

```diff
@@ -65,7 +65,9 @@ let simplify act =
       :: acc
     | System x -> Sh x :: acc
     | Bash x -> Run ("bash", [ "-e"; "-u"; "-o"; "pipefail"; "-c"; x ]) :: acc
-    | Write_file (x, y) -> Redirect_out (echo y, Stdout, File x) :: acc
+    | Write_file (x, y, b) ->
+      assert b;
```

Comment:
Could you leave a comment as to why this is allowed?

Reply:
As long as I've correctly understood it myself - I think I'm correct that this is only reached with actions which will have been generated from dune files? The alternative is to ignore the `b` parameter, but I wasn't liking that!

Reply:
>  I think I'm correct that this is only reached with actions which will have been generated from dune files? 

I suppose this is why I was confused by this assertion in the first place. `Action_to_sh` is used to implement rule printing by `$ dune rules -m`. This should print all rules and not just those generated from dune files.